test/e2e/cmd/clusters.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"fmt"
+	"os/exec"
+	"os/user"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-container-networking/test/e2e/framework/azure"
+	"github.com/Azure/azure-container-networking/test/e2e/framework/types"
+	"github.com/spf13/cobra"
+)
+
+func newClusterCmd() *cobra.Command {
+	clusterCmd := &cobra.Command{
+		Use:   "cluster",
+		Short: "deploys a cluster",
+		CompletionOptions: cobra.CompletionOptions{
+			DisableDefaultCmd: true,
+		},
+	}
+
+	clusterCmd.AddCommand(newBYOCiliumCmd())
+
+	return clusterCmd
+}
+
+func newBYOCiliumCmd() *cobra.Command {
+	byocilium := &cobra.Command{
+		Use:   "byocilium",
+		Short: "deploys a BYO Cilium Cluster",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			job := types.NewJob("deploy BYO Cilium Cluster")
+
+			sub, err := GetCurrentAzCLISubscriptionID()
+			if err != nil {
+				return fmt.Errorf("failed to get subscription id: %w", err)
+			}
+
+			curuser, _ := user.Current()
+			clusterName := curuser.Username + "-byocilium-" + strconv.FormatInt(time.Now().Unix(), 10)
+
+			job.AddStep(&azure.CreateResourceGroup{
+				SubscriptionID:    sub,
+				ResourceGroupName: clusterName,
+				Location:          "westus2",
```

Comment:
nit: Shouldn't we make names and locations dynamic?

Reply:
yes but let's get this is so we can start using it and learning what could be improved like this